Extension/src/LanguageServer/configurations.ts

```diff
@@ -393,11 +393,18 @@ export class CppProperties {
         return result;
     }
 
+    private get additionalEnvironment(): { [key: string]: string | string[] } {
+        let env: { [key: string]: string | string[] } = this.configurationJson.env;
+        env["workspaceFolderBasename"] = path.basename(this.rootUri.fsPath);
```

Comment:
>  think it would be better to just add "workspaceFolderBasename" to configurationJson.env right after it is parsed 

I considered that first but then I thought that it sounded wrong, semantically. I mean, one might need a way to get the actual environment that was set by the user in the configuration json. Thoughts?

> We should also update c_cpp_properties.schema.json's patternProperties to disallow workspaceFolderBasename as a property of env so that green squiggles will appear in c_cpp_properties.json if anyone tries to set it manually.

Definitely! I'm on it.

Reply:
> so you're modifying the .env here anyway

Yeah, you are right. 

> And if we do it in parsePropertiesFile() we only have to do it once every time

True, let's go with that... One possible caveat: will `parsePropertiesFile()` be called again if the folder is renamed?